projecttacoma/fqm-execution
# Valuesets Fix

[bundles.zip](https://github.com/projecttacoma/fqm-execution/files/11324588/bundles.zip)
# Summary
Fqm-execution was only pulling in valuesets listed in Library dataRequirements codeFilters, but needs all valueset relatedArtifacts as well to properly supply results. This PR makees updates to ValueSetHelper.ts to retrieve valuesets from VSAC included in the relatedArtifacts of a Library that are not included in the MeasureBundle

## New behavior
Calculation on MeasureBundles that do not include all necessary valuesets will now pull in valuesets listed in the `relatedArtifacts` property of included Libraries

## Code changes
Added additional checks to `getMissingDependentValuesets` that look for valuesets listed in `relatedArtifacts`

# Testing guidance
- `npm run check`
- run `npm run cli -- -p "issuePB.json" -m "issueMB.json" -o --vs-api-key {VSAC_API_KEY}` on the attached bundles in master and see that the patient incorrectly evaluates outside of the `IPP`
- run the same on my branch and see the patient calculate into the numerator
- run `npm run cli -- valueSets -m "issueMB.json" -o --vs-api-key {VSAC_API_KEY}` and ensure that the output bundle has 26 valueset resources.
- Please reach out if you need a VSAC API key for testing purposes!
